### DISCUSSION TITLE
Server cannot interact with a Docker container?

### DISCUSSION THREAD

**Discussion by joshuarobs** [community/community#25051] [Actions]
<p>I’m not sure if this is a bug with Hasura (<a href="https://github.com/hasura/graphql-engine" rel="nofollow noopener">https://github.com/hasura/graphql-engine</a>), or with the Github servers, or even me, but I can’t seem to get a Hasura cli (<a href="https://www.npmjs.com/package/hasura-cli" rel="nofollow noopener">https://www.npmjs.com/package/hasura-cli</a>) command work with a setup docker container.</p>
<p>This is what the workflow file looks like:</p>
<pre><code>name: Node CI

on: [push]

jobs:
  build:

    runs-on: ubuntu-16.04

    strategy:
      matrix:
        node-version: [12.x]

    steps:
    - uses: actions/checkout@v1
    - name: Install Hasura cli
      run: curl -L https://github.com/hasura/graphql-engine/raw/master/cli/get.sh | bash
    - name: Start up the Docker containers and migrate to the latest version of the database
      run: |
        docker-compose up -d
        sleep 4
        cd docker
        echo $(docker ps)
        hasura migrate status --skip-update-check --endpoint http://localhost:8080
    - name: Use Node.js ${{ matrix.node-version }}
      uses: actions/setup-node@v1
      with:
        node-version: ${{ matrix.node-version }}
    - name: npm install, build, and test
      run: |
        node index.js
      env:
        CI: true
</code></pre>
<p>The line <code>hasura migrate status --skip-update-check --endpoint http://localhost:8080</code> is what fails. If I run this code in the same repo locally on my machine (macOS Catalina latest), it works fine.</p>
<p>I made a test repo that you can clone and see for yourself:</p>
<p><a href="https://github.com/joshuarobs/making-hasura-work-with-github-actions" rel="nofollow noopener">https://github.com/joshuarobs/making-hasura-work-with-github-actions</a></p>
<p>Simply just clone it, push something to master and hopefully it starts automatically doing the tests.</p>
<p>Here’s a link to the latest build run log which failed: <a href="https://github.com/joshuarobs/making-hasura-work-with-github-actions/runs/375147959" rel="nofollow noopener">https://github.com/joshuarobs/making-hasura-work-with-github-actions/runs/375147959</a></p>
<p>I’ve also tried other variations in the endpoint such as:</p>
<p>* empty, i.e. removing the <code>--endpoint</code> flag</p>
<p>* https</p>
<pre><code>https://localhost:8080
</code></pre>
<p>* graphql endpoint</p>
<pre><code>http://localhost:8080/v1/graphql
</code></pre>
<p>But none of these have worked.</p>
<p>* Related issue on Hasura: <a href="https://github.com/hasura/graphql-engine/issues/3638" rel="nofollow noopener">https://github.com/hasura/graphql-engine/issues/3638</a></p>
<p>Is it just me, or is there something wrong?</p>

**Comment by joshuarobs** [comment]
<p>After doing some more brute forcing, to find the root cause of the problem, I think I’ve finally found a lead.</p>
<p>I ran pg_dump just to see what’s in each database and found a discrepancy. This is the command I used:</p>
<pre><code>docker exec container-name_actions_postgres_1 pg_dump -U postgres -d postgres
</code></pre>
<p>The container name is obtained from <code>docker ps</code> and looking at the one that ends with <code>postgres_1</code>. The actual full command, as seen in the repo is:</p>
<pre><code>docker exec making-hasura-work-with-github-actions_postgres_1 pg_dump -U postgres -d postgres
</code></pre>
<p>It runs automatically in the build. Running that locally and manually will also work.</p>
<p>Here is the <strong>pg_dump I get from GitHub Actions</strong> :</p>
<pre><code>--
-- PostgreSQL database dump
--

-- Dumped from database version 12.1 (Debian 12.1-1.pgdg100+1)
-- Dumped by pg_dump version 12.1 (Debian 12.1-1.pgdg100+1)

SET statement_timeout = 0;
SET lock_timeout = 0;
SET idle_in_transaction_session_timeout = 0;
SET client_encoding = 'UTF8';
SET standard_conforming_strings = on;
SELECT pg_catalog.set_config('search_path', '', false);
SET check_function_bodies = false;
SET xmloption = content;
SET client_min_messages = warning;
SET row_security = off;

--
-- PostgreSQL database dump complete
--
</code></pre>
<p>This is where it gets even more interesting, <strong>the pg_dump on my local machine:</strong></p>
<pre><code>--
-- PostgreSQL database dump
--

-- Dumped from database version 12.1 (Debian 12.1-1.pgdg100+1)
-- Dumped by pg_dump version 12.1 (Debian 12.1-1.pgdg100+1)

SET statement_timeout = 0;
SET lock_timeout = 0;
SET idle_in_transaction_session_timeout = 0;
SET client_encoding = 'UTF8';
SET standard_conforming_strings = on;
SELECT pg_catalog.set_config('search_path', '', false);
SET check_function_bodies = false;
SET xmloption = content;
SET client_min_messages = warning;
SET row_security = off;

--
-- Name: hdb_catalog; Type: SCHEMA; Schema: -; Owner: postgres
--

CREATE SCHEMA hdb_catalog;


ALTER SCHEMA hdb_catalog OWNER TO postgres;

--
-- Name: hdb_views; Type: SCHEMA; Schema: -; Owner: postgres
--

CREATE SCHEMA hdb_views;


ALTER SCHEMA hdb_views OWNER TO postgres;

--
-- Name: pgcrypto; Type: EXTENSION; Schema: -; Owner: -
--

CREATE EXTENSION IF NOT EXISTS pgcrypto WITH SCHEMA public;


--
-- Name: EXTENSION pgcrypto; Type: COMMENT; Schema: -; Owner:
--

COMMENT ON EXTENSION pgcrypto IS 'cryptographic functions';


--
-- Name: hdb_schema_update_event_notifier(); Type: FUNCTION; Schema: hdb_catalog; Owner: postgres
--

CREATE FUNCTION hdb_catalog.hdb_schema_update_event_notifier() RETURNS trigger
    LANGUAGE plpgsql
    AS $$
  DECLARE
    instance_id uuid;
    occurred_at timestamptz;
    curr_rec record;
  BEGIN
    instance_id = NEW.instance_id;
    occurred_at = NEW.occurred_at;
    PERFORM pg_notify('hasura_schema_update', json_build_object(
      'instance_id', instance_id,
      'occurred_at', occurred_at
      )::text);
    RETURN curr_rec;
  END;
$$;
</code></pre>
<p>That’s not even the full output pasted. For some reason, the pg_dump in the GitHub Actions machine doesn’t contain all the SQL stuff I’d assume that would make Hasura work.</p>
<p>And this is with all an “empty” Hasura database. Going to <code>http://localhost:8080/console</code> in the Data tab will have no tables. I’m not sure what to do past this point, but I’ll try to find a way to make the cloud machine start up the same as my local dev environment.</p>
<p>Anyone have ideas as to why there is no SQL if this is built on the server?</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/joshuarobs">@joshuarobs</a> ,</p>
<p>I notice you have answered in this <a href="https://github.com/hasura/graphql-engine/issues/3649#issuecomment-572525873" rel="nofollow noopener">link</a>, so the issue is resolved, right? </p>
<p>You are always welcome if you have any Github Action queries here.</p>
<p>Thanks.</p>
